### PR TITLE
Add a new OceanBase Cloud Integration

### PR DESCRIPTION
**Pull request by kyle-oceanbase**
### What does this PR do?

Add a new integration for [OceanBase Cloud](https://en.oceanbase.com/product/cloud).

### Motivation
[OceanBase Cloud](https://en.oceanbase.com/product/cloud) provides fully managed database services with elastic scalability, ultra-fast performance, and high compatibility on global cloud infrastructure.

We Use the OceanBase Cloud Datadog integration to export metrics from OceanBase Cloud clusters to Datadog

This is a special agent based integration, which should only be installed in OceanBase Cloud environments and users do not need to install by themselves. So we do not submmit our source code or integration configuration file

What inspired you to submit this pull request?
Our users want to use datadog to monitor their OceanBase Cloud Clusters.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)
- [ ] If this PR includes a log pipeline, please add a description describing the remappers and processors.

### CODE REVIEW

**Comment by cswatt** [comment]
Thanks! Tracking documentation review with DOCS-8214.

File: oceanbasecloud/metadata.csv [hunk 1]
```diff
@@ -0,0 +1,16 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name,curated_metric
+oceanbasecloud.load1,gauge,,,,Past 1 minute system average load,1,oceanbasecloud,,
```
Comment (buraizu) [inline oceanbasecloud/metadata.csv:2]:
```suggestion
oceanbasecloud.load1,gauge,,,,Average load over the past one minute,1,oceanbasecloud,,
```

File: oceanbasecloud/metadata.csv [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+oceanbasecloud.cpu_percent,gauge,,percent,,CPU Percent,1,oceanbasecloud,,
```
Comment (buraizu) [inline oceanbasecloud/metadata.csv:3]: Could you please clarify whether this is the percentage of CPU being _utilized_ or _free_?

File: oceanbasecloud/metadata.csv [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+oceanbasecloud.memory_used,gauge,,percent,,Used physical memory percent,1,oceanbasecloud,,
+oceanbasecloud.net_throughput,gauge,,mebibyte,,Network throughput,1,oceanbasecloud,,
+oceanbasecloud.ob_data_disk_percent,gauge,,percent,,OB Data Disk Usag,1,oceanbasecloud,,
```
Comment (buraizu) [inline oceanbasecloud/metadata.csv:6]:
```suggestion
oceanbasecloud.ob_data_disk_percent,gauge,,percent,,OB Data Disk Usage,1,oceanbasecloud,,
```

File: oceanbasecloud/metadata.csv [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+oceanbasecloud.ob_clog_disk_percent,gauge,,percent,,OB Clog Disk Usage,1,oceanbasecloud,,
+oceanbasecloud.ob_process_exists,gauge,,,,OB process exists status,1,oceanbasecloud,,
```
Comment (buraizu) [inline oceanbasecloud/metadata.csv:8]:
```suggestion
oceanbasecloud.ob_process_exists,gauge,,,,OB process existence status,1,oceanbasecloud,,
```

File: oceanbasecloud/metadata.csv [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+oceanbasecloud.sql_all_count,gauge,,execution,,Number of SQL statements processed per second,1,oceanbasecloud,,
+oceanbasecloud.sql_all_rt,gauge,,millisecond,,The average processing time of each SQL statement,1,oceanbasecloud,,
+oceanbasecloud.all_session,gauge,,,,Number of current sessions,1,oceanbasecloud,,
+oceanbasecloud.request_queue_time,gauge,,microsecond,,Request waiting queue time consuming,1,oceanbasecloud,,
```
Comment (buraizu) [inline oceanbasecloud/metadata.csv:12]:
```suggestion
oceanbasecloud.request_queue_time,gauge,,microsecond,,Time spent waiting in the request queue,1,oceanbasecloud,,
```

Comment (buraizu) [reply oceanbasecloud/metadata.csv:12]: Suggesting a minor clarification, please confirm if this update is still accurate

File: oceanbasecloud/assets/dashboards/oceanbasecloud_overview.json [hunk 6]
```diff
@@ -0,0 +1,665 @@
...
+        "layout": {
+          "x": 0,
+          "y": 4,
+          "width": 6,
+          "height": 4
+        }
+      }, {
+        "id": 5889700880357092,
+        "definition": {
+          "title": "Request waiting queue time consuming",
```
Comment (buraizu) [inline oceanbasecloud/assets/dashboards/oceanbasecloud_overview.json:127]:
```suggestion
          "title": "Time spent waiting in the request queue",
```

File: oceanbasecloud/assets/dashboards/oceanbasecloud_overview.json [hunk 7]
```diff
@@ -0,0 +1,665 @@
...
+        "layout": {
+          "x": 6,
+          "y": 8,
+          "width": 6,
+          "height": 4
+        }
+      }, {
+        "id": 353434617129980,
+        "definition": {
+          "title": "OB process exists status",
```
Comment (buraizu) [inline oceanbasecloud/assets/dashboards/oceanbasecloud_overview.json:594]:
```suggestion
          "title": "OB process existence status",
```

File: oceanbasecloud/assets/dashboards/oceanbasecloud_overview.json [hunk 8]
```diff
@@ -0,0 +1,665 @@
...
+        "layout": {
+          "x": 0,
+          "y": 8,
+          "width": 6,
+          "height": 4
+        }
+      }, {
+        "id": 2391245474255482,
+        "definition": {
+          "title": "OB Data Disk Usag",
```
Comment (buraizu) [inline oceanbasecloud/assets/dashboards/oceanbasecloud_overview.json:553]:
```suggestion
          "title": "OB Data Disk Usage",
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for the PR, I've suggested some minor edits to the metric descriptions and associated dashboard titles

File: oceanbasecloud/metadata.csv (hunk 2 above)
Comment (kyle-oceanbase) [reply oceanbasecloud/metadata.csv:3]: this is the percentage of CPU being utilized

File: oceanbasecloud/README.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+## Overview
+
+[OceanBase Cloud][1] provides fully managed database services with elastic scalability, ultra-fast performance, and high compatibility on global cloud infrastructure.
+
+Use the OceanBase Cloud Datadog integration to export metrics from OceanBase Cloud clusters to Datadog.
+
+## Setup
+
+To set up the OceanBase Cloud Datadog integration for your cluster, provide a Datadog API key and region to OceanBase Cloud.
```
[the remark this answers is not in the document]
Comment (buraizu) [reply oceanbasecloud/README.md:11]: Suggesting an update here for internal consistency with the rest of the documentation site

File: oceanbasecloud/metadata.csv [hunk 10]
```diff
@@ -0,0 +1,16 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name,curated_metric
+oceanbasecloud.load1,gauge,,,,Average load over the past one minute,1,oceanbasecloud,,
+oceanbasecloud.cpu_percent,gauge,,percent,,CPU Percent,1,oceanbasecloud,,
```
Comment (buraizu) [inline oceanbasecloud/metadata.csv:3]:
```suggestion
oceanbasecloud.cpu_percent,gauge,,percent,,Percentage of CPU in use,1,oceanbasecloud,,
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for following up with my question and applying those changes. I've made one additional suggestion based on your feedback, and also added an additional minor suggestion.

**Review by DrkSephy — APPROVED** [review]

**Comment by emarsha94** [comment]
@kyle-oceanbase Might be best to close this PR and open a new one given that the steps for an API integration are different. I will go ahead and close this out for now!

**Pull request closed without merging**